scrapy/linkextractors/lxmlhtml.py

```diff
@@ -52,6 +52,12 @@ def _iter_links(self, document):
                     continue
                 yield (el, attrib, attribs[attrib])
 
+    def check_for_invalid_sublink (self, url):
+            invalid_url_substring = "../"
+            if invalid_url_substring in url:
+                url = url.replace("../", "")
+            return url
```

Comment:
I'd like for it to be more conservative:

1) shouldn't we only check for '../' in path? Can it appear e.g. in query or in fragment?
2) if Python 3 works correctly, it makes sense to remove a check, to make things faster;
3) is it possible to have more than one '../' in URL? Is it possible to have them in any part of a path, not only right after the hostname? If no, it'd be good to be more conservative and only e.g. remove '../' after the hostname.

Reply:
> is it possible to have more than one '../' in URL?

@kmike I think it is possible if a static website has this kind of structure:
&nbsp;&nbsp;file1.html
&nbsp;&nbsp;folder1/folder2/file2.html

and they are trying to make a link to file1.html in file2.html. In this case, they would have to use **../../** , right?
So I think the worst case is the url in which it contains multiple **../** after the hostname. 

And what is the query and fragment that you were mentioning about? Please let me know! I am trying to make this solution available in w3lib instead!